src/tbb/arena.h

```diff
@@ -179,6 +179,76 @@ class atomic_flag {
     }
 };
 
+#if __TBB_PREVIEW_PARALLEL_BLOCK
+class thread_leave_manager {
+    static const std::uint64_t FAST_LEAVE = 1;
+    static const std::uint64_t ONE_TIME_FAST_LEAVE = 1 << 1;
+    static const std::uint64_t DELAYED_LEAVE = 1 << 2;
+    static const std::uint64_t PARALLEL_BLOCK = 1 << 3;
+    static const std::uint64_t PARALLEL_BLOCK_MASK = ~((1LLU << 32) - 1) & ~(0x7);
```

Comment:
Could you please describe in comments what bits are supposed to be set and unset after this operation?

Also, the following mask is equivalent:
```suggestion
    // The PARALLEL_BLOCK_MASK has the most significant 32 bits set
    static const std::uint64_t PARALLEL_BLOCK_MASK = ~((1LLU << 32) - 1);
```

Reply:
` & ~(0x7)` should be used to nullify state bits

Reply:
I believe in `~((1LLU << 32) - 1)` these bits are already nullified, hence applying `& ~(0x7)` does not change anything. What did I miss?

Reply:
Yes. You are correct. However, the current mask is not 😄 
The mask should be `(1LLU << 32) - 1) & ~(0x7)`

Reply:
Yeah, it is incorrect mask. The correct one should be `((1LLU << 32) - 1) & ~(0x7)`.